key.ui/src/main/java/de/uka/ilkd/key/gui/TaskTree.java

```diff
@@ -73,10 +75,77 @@ public TaskTree(KeYMediator mediator) {
         delegateView.setShowsRootHandles(false);
         delegateView.setRootVisible(false);
         delegateView.putClientProperty("JTree.lineStyle", "Horizontal");
+
+        // create a context menu on demand
+        delegateView.addMouseListener(this);
```

Comment:
There already is another mouse listener (see field "mouseListener"). Is it necessary to have two? Can they be merged? In addition, I think it is cleaner to use inner classes instead of having TaskTree implement the MouseListener/PopupMenuListener interfaces.

Reply:
This is done.